examples/vision/video_transformers.py

```diff
@@ -0,0 +1,408 @@
+"""
+
+"""shell
+wget -q https://git.io/JZmf4 -O top5_data_prepared.tar.gz
+tar xf top5_data_prepared.tar.gz
+"""
+
+train_data, train_labels = np.load("train_data.npy"), np.load("train_labels.npy")
+test_data, test_labels = np.load("test_data.npy"), np.load("test_labels.npy")
+
+print(f"Frame features in train set: {train_data.shape}")
+
+"""
+## Building the Transformer-based model
+
+We will be building on top of the code shared in 
+[this book chapter](https://livebook.manning.com/book/deep-learning-with-python-second-edition/chapter-11) of 
+[Deep Learning with Python (Second ed.)](https://www.manning.com/books/deep-learning-with-python)
+by François Chollet. 
+
+First, self-attention layers that form the basic blocks of a Transformer are
+order-agnostic. Since, videos are ordered sequences of frames we would need our
+Transformer model to account for that. This is done via **positional encodings**. We will
+simply embed the positions of the frames present inside videos with
+[`layers.Embedding()`](https://keras.io/api/layers/core_layers/embedding). We will then
+add these positional embeddings to the precomputed CNN feature maps. 
+"""
+
+
+class PositionalEmbedding(layers.Layer):
+    def __init__(self, sequence_length, output_dim, **kwargs):
+        super().__init__(**kwargs)
+        self.position_embeddings = layers.Embedding(
+            input_dim=sequence_length, output_dim=output_dim
+        )
+        self.sequence_length = sequence_length
+        self.output_dim = output_dim
+
+    def call(self, inputs):
+        # Our inputs are of shape: `(batch_size, frames, num_features)
+        length = tf.shape(inputs)[1]
+        positions = tf.range(start=0, limit=length, delta=1)
+        embedded_positions = self.position_embeddings(positions)
+        return inputs + embedded_positions
+
+    def compute_mask(self, inputs, mask=None):
+        return tf.math.not_equal(inputs, 0)
+
+    def get_config(self):
+        config = super(PositionalEmbedding, self).get_config()
+        config.update(
+            {"output_dim": self.output_dim, "sequence_length": self.sequence_length,}
+        )
+        return config
+
+
+"""
+Now, we can create a subclassed layer for the Transformer. 
+"""
+
+
+class TransformerEncoder(layers.Layer):
+    def __init__(self, embed_dim, dense_dim, num_heads, **kwargs):
+        super().__init__(**kwargs)
+        self.embed_dim = embed_dim
+        self.dense_dim = dense_dim
+        self.num_heads = num_heads
+        self.attention = layers.MultiHeadAttention(
+            num_heads=num_heads, key_dim=embed_dim, dropout=0.3
+        )
+        self.dense_proj = keras.Sequential(
+            [layers.Dense(dense_dim, activation=tf.nn.gelu), layers.Dense(embed_dim),]
+        )
+        self.layernorm_1 = layers.LayerNormalization()
+        self.layernorm_2 = layers.LayerNormalization()
+
+    def call(self, inputs, mask=None):
```

Comment:
Also no need for all the mask-related code, you don't use padding/masking in this model